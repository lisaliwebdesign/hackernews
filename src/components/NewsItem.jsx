import React from "react";
import Moment from "react-moment";
import PropTypes from "prop-types";
import { NewsItemContainer } from "../elements/news";

export default class NewsItem extends React.Component {
  static propTypes = {
    data: PropTypes.instanceOf(Object)
  };

  render() {
    return <NewsItemContainer content={this.props.data} />;
  }
}
