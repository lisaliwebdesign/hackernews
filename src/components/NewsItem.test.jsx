import NewsItem from "./NewsItem";

it("<components><NewsItems - renders component", () => {
  const testingData = [
    {
      by: "rlancer",
      descendants: 399,
      id: 18428497,
      kids: [18429795, 18429547],
      score: 773,
      time: 1541969277,
      title:
        "Google Kubernetes Engine's third consecutive day of service disruption",
      type: "story",
      url: "https://status.cloud.google.com/incident/container-engine/18005"
    },
    {
      by: "rlancer",
      descendants: 399,
      id: 18428497,
      kids: [18429795, 18429547],
      score: 773,
      time: 1541969277,
      title:
        "Google Kubernetes Engine's third consecutive day of service disruption",
      type: "story",
      url: "https://status.cloud.google.com/incident/container-engine/18005"
    }
  ];
  const type = "beststories";

  const wrapper = shallow(<NewsItem content={testingData} type={type} />);
  expect(wrapper).toMatchSnapshot();
});
