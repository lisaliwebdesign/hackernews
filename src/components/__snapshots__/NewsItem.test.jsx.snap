// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<components><NewsItems - renders component 1`] = `
<NewsItemContainer
  type="beststories"
/>
`;
